hyperjumptech/monika
# Allow unauthenticated SMTP

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Implements #1012 
2. Remove mandatory username/password for smtp channel

## How did you implement / how did you fix it  
1.  make username/password optional in validator
2. make username/password mandatori in json schema

## How to test  
1. Using [Mailcatcher](https://mailcatcher.me/)
2. Use this docker-compose to run a simple smtp locally
```yaml
version: "3"

services:
  mailcatcher:
    restart: on-failure
    image: dockage/mailcatcher:0.8.2
    container_name: mailcatcher
    ports:
    - "1080:1080"
    - "1025:1025"
```
3. type `docker comose up` to run the mailcatcher
4. Next use this notification setting is your monika config.yaml
```yaml
notifications:
  - id: smtp-mailcatcher
    type: smtp
    data:
      recipients: ['budhi@hyperjump.tech']
      hostname: 0.0.0.0
      port: 1025
```
5. Now run monika with `bin/run-dev -c config.yaml` above. 
6. Expect everything to run normally and mail to be caught in the mailcatcher, check port 1080

### Mail notifications on mailcatcher
![image](https://user-images.githubusercontent.com/964106/227437930-1c081d5d-17cb-49c6-8cdd-57a097414575.png)

Monika running
![image](https://user-images.githubusercontent.com/964106/227437363-ca5b8fa0-f2f8-4843-8d85-0c34ea6d72ea.png)
